### DISCUSSION TITLE
3.8.0: Getting a g++ error when compiling server.cpp

### DISCUSSION THREAD

**Discussion by mchch007** [jamulussoftware/jamulus#2052] [Bug Investigation]
Apologies upfront if this is already known; I am new here.

If am doing anything wrong instead, as the problem is often sitting in front of the machine, just let me know.

Here is my description:

When compiling in a Ubuntu 20.04 environment, which is also the target one, I am getting a g++ error.
Previous compilations in the same environment went well.

My machine: 
Linux harry.ddnss.de 5.11.0-34-lowlatency #36~20.04.1-Ubuntu SMP PREEMPT Fri Aug 27 09:17:59 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

Sources:
https://github.com/jamulussoftware/jamulus/archive/r3_8_0.tar.gz

Command:
g++ -c -pipe -O2 -std=gnu++11 -D_REENTRANT -Wall -W -fPIC -DAPP_VERSION=\"3.8.0\" -DCUSTOM_MODES -D_REENTRANT -DQT_NO_DEPRECATED_WARNINGS -DHAVE_LRINTF -DHAVE_STDINT_H -DSERVER_ONLY -DHEADLESS -DOPUS_X86_MAY_HAVE_SSE -DOPUS_X86_MAY_HAVE_SSE2 -DOPUS_X86_MAY_HAVE_SSE4_1 -DOPUS_X86_PRESUME_SSE=1 -DOPUS_X86_PRESUME_SSE2=1 -DCPU_INFO_BY_C -DOPUS_BUILD=1 -DUSE_ALLOCA=1 -DOPUS_HAVE_RTCD=1 -DHAVE_LRINTF=1 -DHAVE_LRINT=1 -DQT_NO_DEBUG -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_CONCURRENT_LIB -DQT_CORE_LIB -I. -Isrc -Ilibs/opus/include -Ilibs/opus/celt -Ilibs/opus/silk -Ilibs/opus/silk/float -Ilibs/opus/silk/fixed -Ilibs/opus -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtXml -isystem /usr/include/x86_64-linux-gnu/qt5/QtConcurrent -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I. -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -o server.o src/server.cpp

Error message:
{standard input}: Assembler messages:
{standard input}:65104: Warning: end of file not at end of a line; newline inserted
{standard input}:65436: Error: number of operands mismatch for `movq'
{standard input}: Error: open CFI at the end of file; missing .cfi_endproc directive
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [Makefile:2561: server.o] Error 1

**Comment by rdica** [comment]
Can you detail your build steps?

**Reply by rdica** [reply]
I fired up a Ubuntu Focal 20.04 VM, installed the deps, and successfully compiled.

**Reply by mchch007** [reply]
Could you share your  server.cpp file? I'd like to give it a try and/ or compare it to my version.

**Reply by mchch007** [reply]
Build steps are just downloading the sources as described and running makefile. Other cpp are compiled well, just the server.cpp compilation comes back with the error after freeze of some 3 to 4 minutes....

**Reply by rdica** [reply]
I don't know what you mean "running makefile".
The build steps are:
- cd $SRC_DIR
- qmake Jamulus.pro
- make clean && make

These are the steps as outlined on github. Are these the steps you followed as well?

**Reply by mchch007** [reply]
Sorry for not having been clear. I did exactly those steps.

**Comment by atsampson** [comment]
This looks like GCC is running out of memory and getting killed - the errors from the assembler are just because GCC gave up writing the assembly output partway through. How much memory does your VM have?

**Reply by mchch007** [reply]
Thanks for notifying. My small VM has 1 GB. I will setup a bigger one and retry. Will be able to do this only during the weekend. Will come back with the result.

**Reply by mchch007** [reply]
Wow, that was it!  Thanks so much!
As my server is also running as a small "production" for Jamulus, I wanted to avoid rebuilding the whole thing, and instead could free up some 300M memory by stopping a second application not needed anymore. This have me some 450M available memory
(MiB Mem :    976.6 total,    431.6 free,    129.4 used,    415.7 buff/cache), and that was it for compiling.
Again, thanks for your valuable answers and support!
